docs/docs/09.1-language-tooling.md

```diff
@@ -0,0 +1,79 @@
+
+> Note:
+>
+> The in-browser JSX transformer is fairly large and results in extraneous computation client-side that can be avoided. Do not use it in production — see the next section.
+
+### Productionizing: Precompiled JSX
+
+If you have [npm](https://www.npmjs.com/), you can run `npm install -g babel-cli`. Babel has built-in support for React v0.12+. Tags are automatically transformed to their equivalent `React.createElement(...)`, `displayName` is automatically inferred and added to all `React.createClass` calls.
+
+This tool will translate files that use JSX syntax to plain JavaScript files that can run directly in the browser. It will also watch directories for you and automatically transform files when they are changed; for example: `babel --watch src/ --out-dir lib/`.
+
+Beginning with Babel 6, there are no transforms included by default. This means that options must be specified when running the `babel` command, or a `.babelrc` must specify options. Additional packages must also be installed which bundle together a number of transforms, called presets. The most common use when working with React will be to include the `es2015` and `react` presets. More information about the changes to Babel can be found in [their blog post announcing Babel 6](http://babeljs.io/blog/2015/10/29/6.0.0).
+
+Here is an example of what you will do if using ES2015 syntax and React:
+
+```
+npm install babel-preset-es2015 babel-preset-react
+babel --presets es2015,react --watch src/ --out-dir lib/
+```
+
+By default JSX files with a `.js` extension are transformed. Run `babel --help` for more information on how to use Babel.
+
+Example output:
+
+```
+$ cat test.js
+var HelloMessage = React.createClass({
+  render: function() {
+    return <div>Hello {this.props.name}</div>;
+  }
+});
+
+ReactDOM.render(<HelloMessage name="John" />, mountNode);
+$ babel test.js
+"use strict";
+
+var HelloMessage = React.createClass({
+  displayName: "HelloMessage",
+
+  render: function render() {
+    return React.createElement(
+      "div",
+      null,
+      "Hello ",
+      this.props.name
+    );
+  }
+});
+
+ReactDOM.render(React.createElement(HelloMessage, { name: "John" }), mountNode);
+```
+
+### Helpful Open-Source Projects
+
+The open-source community has built tools that integrate JSX with several editors and build systems. See [JSX integrations](https://github.com/facebook/react/wiki/Complementary-Tools#jsx-integrations) for the full list.
+
+## Flow
+
+Flow is a JavaScript type checker released by Facebook, and it supports JSX.  For more info, checkout the [Flow homepage](http://flowtype.org/).
+
+
+## TypeScript
+
+Microsoft TypeScript now supports JSX.  For more info, check out the [TypeScript Wiki](https://github.com/Microsoft/TypeScript/wiki/JSX).
```

Comment:
Linking to our site's documentation might be better: http://www.typescriptlang.org/docs/handbook/jsx.html